test/snowpack/protocol_test.exs

```diff
@@ -1,36 +1,51 @@
 defmodule Snowpack.ProtocolTest do
-  use ExUnit.Case, async: true
+  use ExUnit.Case, async: false
 
   use Mimic
 
   import Snowpack.TestHelper
 
   describe "handle_execute/4" do
-    setup [:connect]
+    setup [:connect, :set_mimic_global]
 
+    @tag :capture_log
     test "with connection errors", %{pid: pid} do
-      expect(Snowpack.ODBC, :query, fn _pid, _statement, _params, _opts, _with_query_id ->
+      expect Snowpack.ODBC, :query, fn _pid, _statement, _params, _opts, _with_query_id ->
         {:error, Snowpack.Error.exception({"08123", "123", "bad"})}
-      end)
+      end
 
       assert {:error, %Snowpack.Error{odbc_code: :connection_exception}} = Snowpack.query(pid, "select 1;")
     end
 
+    @tag :capture_log
+    test "with connection_closed error ODBC is disconnected", %{pid: pid} do
+      expect Snowpack.ODBC, :query, fn _pid, _statement, _params, _opts, _with_query_id ->
+        {:error, Snowpack.Error.exception(:connection_closed)}
+      end
+
+      expect :odbc, :disconnect, fn _pid -> :ok end
+
+      assert {:error, %Snowpack.Error{message: "connection_closed"}} = Snowpack.query(pid, "select 1;")
+
+      # HACK: wait for DBConnection to process the disconnect
+      Process.sleep(500)
```

Comment:
😬

Reply:
I might be able to use a connection listener.